src/main/scala/com/redhat/et/silex/rdd/drop/drop.scala

```diff
@@ -0,0 +1,190 @@
+
+    new RDD[T](self.context, List(new OneToOneDependency(self), new FanInDep(locRDD))) {
+      override def getPartitions: Array[Partition] = 
+        self.partitions.map(p => new PromiseArgPartition(p, List(locRDD)))
+
+      override val partitioner = self.partitioner
+
+      override def compute(split: Partition, ctx: TaskContext):Iterator[T] = {
+        val dp = split.asInstanceOf[PromiseArgPartition]
+        val (pFirst, pDrop) = dp.arg[(Int,Int)](0, ctx)
+        val input = firstParent[T]
+        if (dp.index > pFirst) return input.iterator(dp.partition, ctx)
+        if (dp.index == pFirst) return input.iterator(dp.partition, ctx).drop(pDrop)
+        Iterator.empty
+      }
+    }
+  }
+
+  /** Obtain a new RDD formed by dropping the last (n) elements of the input RDD
+    *
+    * @param n The number of rows to drop
+    * @return an RDD formed from dropping the last 'n' rows of the input
+    */
+  def dropRight(n: Int):RDD[T] = {
+    if (n <= 0) return self
+
+    val locate = (partitions: Array[Partition], input: RDD[T], ctx: TaskContext) => {
+      var rem = n
+      var p = partitions.length-1
+      var np = 0
+      while (rem > 0  &&  p >= 0) {
+        np = input.iterator(partitions(p), ctx).length
+        rem -= np
+        p -= 1
+      }
+
+      if (rem > 0  ||  (rem == 0  &&  p < 0)) {
+        // all elements were dropped
+        (p, 0)
+      } else {
+        // (if we get here, note that rem <= 0)
+        (p + 1, -rem)
+      }
+    }
+
+    val locRDD = self.promiseFromPartitionArray(locate).asInstanceOf[PromiseRDD[(Int,Int)]]
+
+    new RDD[T](self.context, List(new OneToOneDependency(self), new FanInDep(locRDD))) {
+      override def getPartitions: Array[Partition] = 
+        self.partitions.map(p => new PromiseArgPartition(p, List(locRDD)))
+
+      override val partitioner = self.partitioner
+
+      override def compute(split: Partition, ctx: TaskContext):Iterator[T] = {
+        val dp = split.asInstanceOf[PromiseArgPartition]
+        val (pFirst, pTake) = dp.arg[(Int,Int)](0, ctx)
+        val input = firstParent[T]
+        if (dp.index < pFirst) return input.iterator(dp.partition, ctx)
+        if (dp.index == pFirst) return input.iterator(dp.partition, ctx).take(pTake)
+        Iterator.empty
+      }
+    }
+  }  
+
+  /** Obtain a new RDD formed by dropping leading rows until predicate function (f) returns false
+    *
+    * @param f Predicate function.  Input rows are dropped until f returns false
+    * @return An RDD formed by dropping leading rows until predicate function (f) returns false
+    */
+  def dropWhile(f: T=>Boolean):RDD[T] = {
+
+    val locate = (partitions: Array[Partition], input: RDD[T], ctx: TaskContext) => {
+      var p = 0
+      var np = 0
+      while (np <= 0  &&  p < partitions.length) {
+        np = input.iterator(partitions(p), ctx).dropWhile(f).length
+        p += 1
+      }
+
+      if (np <= 0  &&  p >= partitions.length) {
```

Comment:
Don't think so, my use of `<=` or `>=` instead of `==` is mostly a defensive habit